ibm/flex/structures.go

```diff
@@ -930,6 +930,130 @@ func ReplicationRuleGet(in *s3.ReplicationConfiguration) []map[string]interface{
 	return rules
 }
 
+func flattenLifecycleExpiration(expiration *s3.LifecycleExpiration) []interface{} {
+	if expiration == nil {
+		return []interface{}{}
+	}
+
+	m := make(map[string]interface{})
+
+	if expiration.Date != nil {
+		m["date"] = expiration.Date.Format(time.RFC3339)
+	}
+
+	if expiration.Days != nil {
+		m["days"] = int(aws.Int64Value(expiration.Days))
+	}
+
+	if expiration.ExpiredObjectDeleteMarker != nil {
+		m["expired_object_delete_marker"] = aws.Bool(*expiration.ExpiredObjectDeleteMarker)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenNoncurrentVersionExpiration(expiration *s3.NoncurrentVersionExpiration) []interface{} {
+	if expiration == nil {
+		return []interface{}{}
+	}
+
+	m := make(map[string]interface{})
+	if expiration.NoncurrentDays != nil {
+		m["noncurrent_days"] = int(aws.Int64Value(expiration.NoncurrentDays))
+	}
+
+	return []interface{}{m}
+}
+func flattenTransitions(transitions []*s3.Transition) []interface{} {
+	if len(transitions) == 0 {
+		return []interface{}{}
+	}
+
+	var results []interface{}
+
+	for _, transition := range transitions {
+		m := make(map[string]interface{})
+
+		if transition.StorageClass != nil {
+			m["storage_class"] = transition.StorageClass
+		}
+
+		if transition.Date != nil {
+			m["date"] = transition.Date.Format(time.RFC3339)
+		}
+
+		if transition.Days != nil {
+			m["days"] = int(aws.Int64Value(transition.Days))
+		}
+
+		results = append(results, m)
+	}
+	return results
+}
+
+func flattenLifecycleRuleFilter(filter *s3.LifecycleRuleFilter) []interface{} {
+	if filter == nil {
+		return []interface{}{}
+	}
+
+	m := make(map[string]interface{})
+
+	if filter.Prefix != nil {
+		m["prefix"] = aws.String(*filter.Prefix)
+
+	}
+
+	return []interface{}{m}
+}
+
+func flattenAbortIncompleteMultipartUpload(u *s3.AbortIncompleteMultipartUpload) []interface{} {
+	if u == nil {
+		return []interface{}{}
+	}
+
+	m := make(map[string]interface{})
+
+	if u.DaysAfterInitiation != nil {
+		m["days_after_initiation"] = int(aws.Int64Value(u.DaysAfterInitiation))
+	}
+
+	return []interface{}{m}
+}
+
+func LifecylceRuleGet(in []*s3.LifecycleRule) []map[string]interface{} {
```

Comment:
Can we change the input name from in to something else?

Reply:
done